test/Date.test.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import Date from '../js/components/Message/Types/JSObject/Date';
+import data from "./data/Date.js";
+
+describe('Date', () => {
+
+  data.forEach((testObj, testName) => {
+    it(`should render a ${testName}`, () => {
+      const message = testObj.responsePacket.result;
+
+      const rendered = shallow(<Date message={message} />);
+      expect(rendered.text()).toContain(testObj.consoleOutput);
```

Comment:
> Do you think we'd want to restructure the HTML as part of the rewrite?

No, I think we should keep the same markup for most output formats to limit risk.  But for anything tied to XUL (console.table in particular) we'll need to do something else.  Maybe for the table we can start out with plain <table> markup and then enhance that with some kind of widget later.  And then in the future when we integrate Reps that will change the output as well, but I think we shouldn't worry about that part for now.